opm/simulators/timestepping/AdaptiveTimeStepping_impl.hpp

```diff
@@ -242,9 +242,22 @@ namespace Opm {
                     OpmLog::note("Overall linear iterations used: " + std::to_string(substepReport.total_linear_iterations));
                 }
             }
+            catch (const Opm::TooManyIterations& e) {
+                substepReport += solver.failureReport();
+                cause_of_failure = "Solver convergence failure - Iteration limit reached";
+
+                detail::logException(e, solver_verbose_);
+                // since linearIterations is < 0 this will restart the solver
+            }
+            catch (const Opm::LinearSolverProblem& e) {
+                substepReport += solver.failureReport();
+                cause_of_failure = "Linear Solver convergence failure";
+
+                detail::logException(e, solver_verbose_);
+                // since linearIterations is < 0 this will restart the solver
+            }
             catch (const Opm::NumericalProblem& e) {
                 substepReport += solver.failureReport();
-                cause_of_failure = e.what();
```

Comment:
Either this should not be removed, or you should make it a generic "Some numerical problem encountered" cause of failure. As written, cause_of_failure will be empty.